src/main/java/frc/robot/commands/DriveCommands.java

```diff
@@ -53,42 +59,143 @@ private static Translation2d getLinearVelocityFromJoysticks(double x, double y)
+                      Logger.recordOutput("Auto Align/PID/Y At Setpoint", yController.atSetpoint());
+
+                      Logger.recordOutput(
+                          "Auto Align/PID/Rot Value (Rads)", Degrees.of(rotValue).in(Radians));
+                      Logger.recordOutput(
+                          "Auto Align/PID/Rot Setpoint (Rads)",
+                          Degrees.of(rotController.getSetpoint()).in(Radians));
+                      Logger.recordOutput(
+                          "Auto Align/PID/Rot At Setpoint", rotController.atSetpoint());
+
+                      ChassisSpeeds speeds = new ChassisSpeeds(xSpeed, ySpeed, rotValue);
+
+                      // Convert to field Relative Speeds
+                      drive.runVelocity(
+                          ChassisSpeeds.fromFieldRelativeSpeeds(speeds, drive.getRotation()));
+
+                      //   drive.runVelocity(speeds);
+                    })
+                .until(endTrigger))
+        .withName("Go To Pose PID");
   }
 
   public static Command GoToPath(PathPlannerPath targetPath, Set<Subsystem> drive) {
     return new DeferredCommand(
-        () -> AutoBuilder.pathfindThenFollowPath(targetPath, PATH_CONSTRAINTS), drive);
+            () -> AutoBuilder.pathfindThenFollowPath(targetPath, kPathConstraints), drive)
+        .withName("Go To Path");
   }
 
-  public static Command RunVelocity(Drive drive, LinearVelocity velocity, double timeSec) {
-    return Commands.deadline(
-        Commands.waitSeconds(timeSec),
-        Commands.run(
+  public static Command AlignToReef(
+      boolean isRightScore, String cameraName, Drive drive, Vision vision) {
+    return new DeferredCommand(
             () -> {
-              drive.runVelocity(
-                  new ChassisSpeeds(velocity, MetersPerSecond.zero(), DegreesPerSecond.zero()));
+              Pose2d TagPose =
+                  vision
+                      .convertLLPose(LimelightHelpers.getTargetPose3d_RobotSpace(cameraName))
+                      .toPose2d();
+              Pose2d tagPoseRotated =
+                  TagPose.rotateAround(
+                      new Translation2d(), drive.getPose().getRotation().unaryMinus());
+              Pose2d tagPoseFieldRelative =
+                  new Pose2d(
+                      tagPoseRotated.getMeasureX().plus(drive.getPose().getMeasureX()),
+                      tagPoseRotated.getMeasureY().times(-1).plus(drive.getPose().getMeasureY()),
+                      new Rotation2d(
+                          vision
+                              .getTagPose((int) LimelightHelpers.getFiducialID(cameraName))
+                              .getRotation()
+                              .getMeasureZ()));
+              Logger.recordOutput("Auto Align/Tag Pose (RR)", TagPose);
+              Logger.recordOutput("Auto Align/Tag Pose Rotated (RR)", tagPoseRotated);
+              Logger.recordOutput("Auto Align/Tag Pose (FF)", tagPoseFieldRelative);
```

Comment:
Probably don't need these extra logs anymore.

Reply:
I think we keep them. If something start to happen with auto align, better to have it then not.